packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts

```diff
@@ -69,8 +80,6 @@ export class TransformerStackSythesizer extends LegacyStackSynthesizer {
    * addFileAsset
    */
   public addFileAsset(asset: FileAssetSource): FileAssetLocation {
-    assertNotNull(this.boundStack);
-
```

Comment:
this is no longer needed StackSynthesizer.boundStack throws if property is not set.